src/towncrier/test/test_build.py

```diff
@@ -1530,3 +1531,55 @@ def test_orphans_in_non_showcontent_markdown(self, runner):
 
         self.assertEqual(0, result.exit_code, result.output)
         self.assertEqual(expected_output, result.output)
+
+    @with_git_project()
+    def test_uncommitted_files(self, runner, commit):
+        """
+        At build time, it will delete any fragment file regardless of its stage,
+        included files that are not part of the git reporsitory,
+        or are just staged or modified.
+        """
+        # 123 is committed, 124 is modified, 125 is just added, 126 is unknown
+
+        with open("foo/newsfragments/123.feature", "w") as f:
+            f.write("Adds levitation")
+        with open("foo/newsfragments/124.feature", "w") as f:
+            f.write("Extends levitation")
```

Comment:
Maybe add more info to know the purpose of each file.

```suggestion
            f.write("Adds levitation. File committed.")
        with open("foo/newsfragments/124.feature", "w") as f:
            f.write("Extends levitation. File modified in git.")
```